packages/upload-client/test/upload.test.js

```diff
@@ -0,0 +1,266 @@
+        with: account.did(),
+        expiration: Infinity,
+      }),
+    ]
+
+    const service = mockService({
+      upload: {
+        list(invocation) {
+          assert.equal(invocation.issuer.did(), issuer.did())
+          assert.equal(invocation.capabilities.length, 1)
+          const invCap = invocation.capabilities[0]
+          assert.equal(invCap.can, UploadCapabilities.list.can)
+          assert.equal(invCap.with, account.did())
+          return res
+        },
+      },
+    })
+
+    const server = Server.create({ id, service, decoder: CAR, encoder: CBOR })
+    const connection = Client.connect({
+      id,
+      encoder: CAR,
+      decoder: CBOR,
+      channel: server,
+    })
+
+    const list = await Upload.list({ issuer, proofs }, { connection })
+
+    assert.equal(list.count, res.count)
+    assert.equal(list.page, res.page)
+    assert.equal(list.pageSize, res.pageSize)
+    assert(list.results)
+    assert.equal(list.results.length, res.results.length)
+    list.results.forEach((r, i) => {
+      assert.equal(r.carCID.toString(), res.results[i].carCID.toString())
+      assert.equal(r.dataCID.toString(), res.results[i].dataCID.toString())
+      assert.equal(r.uploadedAt, res.results[i].uploadedAt)
+    })
+  })
+
+  it('throws on service error', async () => {
+    const account = await Signer.generate()
+    const issuer = await Signer.generate()
+
+    const proofs = [
+      await UploadCapabilities.list.delegate({
+        issuer: account,
+        audience: id,
+        with: account.did(),
+        expiration: Infinity,
+      }),
+    ]
+
+    const service = mockService({
+      upload: {
+        list: () => {
+          throw new Server.Failure('boom')
+        },
+      },
+    })
+
+    const server = Server.create({ id, service, decoder: CAR, encoder: CBOR })
+    const connection = Client.connect({
+      id,
+      encoder: CAR,
+      decoder: CBOR,
+      channel: server,
+    })
+
+    await assert.rejects(Upload.list({ issuer, proofs }, { connection }), {
+      message: 'failed upload/list invocation',
+    })
+  })
+})
+
+describe('Upload.remove', () => {
+  it('removes an upload', async () => {
+    const account = await Signer.generate()
+    const issuer = await Signer.generate()
+    const car = await randomCAR(128)
+
+    const proofs = [
+      await UploadCapabilities.remove.delegate({
+        issuer: account,
+        audience: id,
+        with: account.did(),
+        expiration: Infinity,
+      }),
+    ]
+
+    const service = mockService({
+      upload: {
+        remove(invocation) {
+          assert.equal(invocation.issuer.did(), issuer.did())
+          assert.equal(invocation.capabilities.length, 1)
+          const invCap = invocation.capabilities[0]
+          assert.equal(invCap.can, UploadCapabilities.remove.can)
+          assert.equal(invCap.with, account.did())
+          assert.equal(String(invCap.nb.root), car.roots[0].toString())
+          return null
```

Comment:
same here, i think we need plan or invocation count assertion for this test to verify that remove was called correctly